### ISSUE TITLE
Docs: Revamp Code Path Analysis docs

### ISSUE THREAD

**Issue by kirkwaiblinger** [eslint/eslint#21189]
### Docs page(s)

https://eslint.org/docs/latest/extend/code-path-analysis

### What documentation issue do you want to solve?

The code path analysis docs currently contain out of date information. They make numerous references to the partially constructed code path graph, for example
```js
/**
			 * This is called at the start of analyzing a code path.
			 * In this time, the code path object has only the initial segment.
			 *
			 * @param {CodePath} codePath - The new code path.
			 * @param {ASTNode} node - The current node.
			 * @returns {void}
			 */
			onCodePathStart(codePath, node) {
				// do something with codePath
			},
```

however, this was changed already back in v9 so that the whole code path graph is available by the time the first event fires: https://eslint.org/docs/latest/use/migrate-to-9.0.0#-code-paths-are-now-precalculated.

Furthermore, there is a lot of awkward/ungrammatical wording, such as:
- "CodePath expresses whole of one code path"
- "Difference from doubly linked list is what there are forking and merging"

### What do you think is the correct solution?

I think the page should be overhauled to have current information, and more natural wording.

### Participation

- [x] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I have a branch in progress with lots of revisions to the wording, happy to submit a PR.

**Comment by mdjermanovic** [comment]
> The code path analysis docs currently contain out of date information. They make numerous references to the partially constructed code path graph

Confirmed, PR to fix this and also improve wording is welcome.
